redbot/cogs/streams/streamtypes.py

```diff
@@ -337,73 +337,56 @@ async def get_data(self, url: str, params: dict = {}) -> Tuple[Optional[int], di
                 return None, {}
 
     async def is_online(self):
-        if not self.id:
-            self.id = await self.fetch_id()
-
-        url = TWITCH_STREAMS_ENDPOINT
-        params = {"user_id": self.id}
+        __, user_profile_data = None, None
+        if self.id is None:
+            __, user_profile_data = await self._fetch_user_profile()
 
-        code, data = await self.get_data(url, params)
-        if code == 200:
-            if not data["data"]:
+        stream_code, stream_data = await self.get_data(
+            TWITCH_STREAMS_ENDPOINT, {"user_id": self.id}
+        )
+        if stream_code == 200:
+            if not stream_data["data"]:
                 raise OfflineStream()
-            self.name = data["data"][0]["user_name"]
-            data = data["data"][0]
-            data["game_name"] = None
-            data["followers"] = None
-            data["view_count"] = None
-            data["profile_image_url"] = None
-            data["login"] = None
-
-            game_id = data["game_id"]
-            if game_id:
-                __, game_data = await self.get_data(
-                    "https://api.twitch.tv/helix/games", {"id": game_id}
-                )
-                if game_data:
-                    game_data = game_data["data"][0]
-                    data["game_name"] = game_data["name"]
-            __, user_data = await self.get_data(
-                "https://api.twitch.tv/helix/users/follows", {"to_id": self.id}
-            )
-            if user_data:
-                followers = user_data["total"]
-                data["followers"] = followers
 
-            __, user_profile_data = await self.get_data(
-                "https://api.twitch.tv/helix/users", {"id": self.id}
+            if user_profile_data is None:
+                __, user_profile_data = await self._fetch_user_profile()
```

Comment:
Sorry, that's my bad...
```suggestion
                user_profile_data = await self._fetch_user_profile()
```